src/procedures/procedure.c

```diff
@@ -130,19 +134,29 @@ bool Procedure_ContainsOutput(const ProcedureCtx *proc, const char *output) {
 	return false;
 }
 
-bool Proc_ReadOnly(const char *proc_name) {
+bool Procedure_ReadOnly(const char *proc_name) {
 	assert(__procedures);
 	ProcGenerator gen = raxFind(__procedures, (unsigned char *)proc_name,
 	  			strlen(proc_name));
 	if(gen == raxNotFound) return false; // Invalid procedure specified, handled elsewhere.
 	/* TODO It would be preferable to be able to determine whether a procedure is read-only
 	 * without creating its entire context; this is wasteful. */
-	ProcedureCtx *ctx = gen(NULL, NULL);
+	ProcedureCtx *ctx = gen();
 	bool read_only = ctx->readOnly;
 	Proc_Free(ctx);
 	return read_only;
 }
 
+const char *Procedure_GetName(const ProcedureCtx *proc) {
+	ASSERT(proc != NULL);
+	return proc->name;
+}
+
+bool Procedure_IsReadOnly(const ProcedureCtx *proc) {
+	ASSERT(proc != NULL);
+	return proc->readOnly;
+}
+
```

Comment:
Since the only caller to those functions is proc_procedures
consider moving them there as auxiliary static functions